### PR TITLE
feat: unsubscribe user from marketing emails

### LINKED ISSUE #159: [FEAT] :sparkles: create new route to unsubscribe from marketing emails
### Description

create new route `GET /api/users/:id/marketing-emails/unsubscribe` by which user will not receive marketing emails any more

we can set user's `recieveMarketingEmails` option to `false` in database to unsubscribe from marketing emails

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by aayushchugh**
### Description

New route to unsubscribe email

### Checklist:

-   [x] This pull request follow our contribution guidelines
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation (postman and wiki)
-   [ ] I have written tests for the code
-   [x] I have updated .env.example file for new .env variables

Closes #159 

### What is new behavior?

image

### Breaking Changes?

Need new .env variable `BASE_URL`

### CODE REVIEW

File: src/routes/user.routes.ts [hunk 1]
```diff
@@ -12,11 +15,16 @@ const userRouter = Router();
 /**
  * This route does following things
  * PATCH -> update user's username
- *
+//
```
Comment (tharun634) [inline src/routes/user.routes.ts]:
```suggestion
 *
```

File: src/schemas/user.schema.ts [hunk 2]
```diff
@@ -100,3 +100,24 @@ export const patchUserSchema = z.object({
  * @author aayushchugh
  */
 export type PatchUserSchema = z.infer<typeof patchUserSchema>;
+
+/**
+ * This schema is used to validate `POST /users/:id/marketing-emails/unsubscribe` request
```
Comment (tharun634) [inline src/schemas/user.schema.ts]:
```suggestion
 * This schema is used to validate `GET /users/:id/marketing-emails/unsubscribe` request
```

File: src/schemas/user.schema.ts [hunk 3]
```diff
@@ -100,3 +100,24 @@ export const patchUserSchema = z.object({
...
+ *
+ * @author aayushchugh
+ */
+export const unsubscribeUserFromMarketingEmailSchema = z.object({
```
Comment (tharun634) [inline src/schemas/user.schema.ts]:
```suggestion
export const getUnsubscribeUserFromMarketingEmailSchema = z.object({
```

Not sure if we should do this, just to stay consistent

**Review by tharun634 — CHANGES_REQUESTED** [review]

**Review by is-it-ayush — APPROVED** [review]
Good Code Ayush!! I'll merge with once the requested changes are made.

**Review by tharun634 — APPROVED** [review]

**Pull request merged**